test/integration/proxy-config-list.test.js

```diff
@@ -7,104 +7,138 @@ describe('Proxy Config List', function () {
         var parent = {},
             list = new ProxyConfigList(parent,
                 [
-                    { server: 'https://proxy.com/', tunnel: true }
+                    { host: 'proxy.com', tunnel: true }
                 ]
             );
-        expect(list.resolve('http://www.google.com/').server.getHost()).to.eql('proxy.com');
-        expect(list.resolve('http://example.org/foo/bar.html').server.getHost()).to.eql('proxy.com');
+        expect(list.resolve('http://www.google.com/').host).to.eql('proxy.com');
+        expect(list.resolve('http://example.org/foo/bar.html').host).to.eql('proxy.com');
     });
 
     it('Assigns, <all_urls> as match pattern and respect the disabled prop in the congfig', function () {
         var parent = {},
             list = new ProxyConfigList(parent,
                 [
-                    { server: 'https://proxy.com/', tunnel: true, disabled: true }
+                    { host: 'proxy.com', tunnel: true, disabled: true }
                 ]
             );
         expect(list.resolve('foo://www.foo/bar')).to.eql(undefined);
     });
 
-    it('Matches any URL that uses the http protocol', function () {
+    it('Do not matches URLs that does not have any valid protocol', function () {
```

Comment:
add a test case where target url does not have any protocol and it should still match if proxy server has http suppo